src/python/pants/backend/python/util_rules/lockfile_metadata.py

```diff
@@ -25,6 +27,19 @@ class LockfileMetadata:
     requirements_invalidation_digest: str
     valid_for_interpreter_constraints: InterpreterConstraints
 
+    @staticmethod
+    def new(
+        requirements_invalidation_digest: str,
+        valid_for_interpreter_constraints: InterpreterConstraints,
+    ) -> LockfileMetadata:
+        """Call the most recent version of the `LockfileMetadata` class to construct a concrete
+        instance.
+
+        This static method should be used in place of the `LockfileMetadata` constructor.
```

Comment:
Can you instead override `__init__` for the dataclass? That's what we normally do.

Also afaict there's no need for this change to happen yet

Reply:
Oh, I had meant to update the calling sites as well, but I guess I can leave this to the now-follow up PR.

Reply:
Okay. I recommend reverting this part of the change, given that it's not used

Reply:
Done